src/bsp/platform/stm32f4_nucleo_wifi/xi_bsp_io_net_stm32f4_nucleo_wifi.c

```diff
@@ -0,0 +1,198 @@
+                                           const uint8_t* buf,
+                                           size_t count )
+{
+    ( void )xi_socket;
+    ( void )buf;
+    *out_written_count = count;
+
+    WiFi_Status_t status = WiFi_MODULE_SUCCESS;
+    status = wifi_socket_client_write(xi_socket, count, (char*) buf);
+
+    if ( status == WiFi_MODULE_SUCCESS ) return XI_BSP_IO_NET_STATE_OK;
+    else return XI_BSP_IO_NET_STATE_ERROR;
+}
+
+xi_bsp_io_net_state_t xi_bsp_io_net_read( xi_bsp_socket_t xi_socket,
+                                          int* out_read_count,
+                                          uint8_t* buf,
+                                          size_t count )
+{
+    ( void )xi_socket;
+    ( void )buf;
+    ( void )count;
+
+    *out_read_count = 0;
+
+    if ( xi_wifi_state == xi_wifi_state_connected )
+    {
+      if ( xi_wifi_buffer.size > 0 )
+      {
+          memcpy( buf , xi_wifi_buffer.bytes , xi_wifi_buffer.size );
+          *out_read_count = xi_wifi_buffer.size;
+          xi_wifi_buffer.size = 0;
+      }
+      else return XI_BSP_IO_NET_STATE_BUSY;
+    }
+    else if ( xi_wifi_state == xi_wifi_state_error )
+    {
+      return XI_BSP_IO_NET_STATE_ERROR;    
+    }
+    else if ( xi_wifi_state == xi_wifi_state_connected )
+    {
+      return XI_BSP_IO_NET_STATE_CONNECTION_RESET;    
+    }
+
+    return XI_BSP_IO_NET_STATE_OK;
+}
+
+xi_bsp_io_net_state_t xi_bsp_io_net_close_socket( xi_bsp_socket_t* xi_socket )
+{
+    WiFi_Status_t status = WiFi_MODULE_SUCCESS;
+    status = wifi_socket_client_close( *xi_socket );
+
+    if ( status == WiFi_MODULE_SUCCESS ) return XI_BSP_IO_NET_STATE_OK;
+    else return XI_BSP_IO_NET_STATE_ERROR;
+}
+
+xi_bsp_io_net_state_t xi_bsp_io_net_select( xi_bsp_socket_events_t* socket_events_array,
+                                            size_t socket_events_array_size,
+                                            long timeout_sec )
+{
+    (void)timeout_sec;
+
+    size_t socket_id = 0;
+
+    if ( xi_wifi_state == xi_wifi_state_connected )
+    {
+
+        for ( socket_id = 0; socket_id < socket_events_array_size; ++socket_id )
+        {
+            xi_bsp_socket_events_t* socket_events = &socket_events_array[socket_id];
+
+            if ( 1 == socket_events->in_socket_want_read )
+            {
+                if ( xi_wifi_buffer.size > 0 ) socket_events->out_socket_can_read = 1;
+            }
+
+            if ( 1 == socket_events->in_socket_want_connect )
+            {
+                socket_events->out_socket_connect_finished = 1;
+            }
+
+            if ( 1 == socket_events->in_socket_want_write )
+            {
+                socket_events->out_socket_can_write = 1;
+            }
+        }
+
+        return XI_BSP_IO_NET_STATE_OK;
+
+    }
+    else
+    {
+      return XI_BSP_IO_NET_STATE_ERROR;
+    }
+  }
+
+  void xi_bsp_io_net_socket_data_received_proxy(uint8_t socket_id, uint8_t * data_ptr, uint32_t message_size, uint32_t chunk_size)
```

Comment:
Can you configure the chunk size ? Is there another API that would allow to check if there is any data and copy it once it's needed ? Usually TLS libraries have pending buffers... I believe that this should also have something similar.